### PR TITLE
nixos/networkd-dispatcher: init at 2.2.4

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Add networkd-dispatcher package and module. To use it, add to `configuration.nix`:

```
services.networkd-dispatcher.enable = true;
``` 

Create a example script `/var/lib/networkd-dispatcher/routable.d/00-restart-tor.sh`
```
#!/usr/bin/env bash
if [[ $IFACE == "wlan0" && $AdministrativeState == "configured" ]]; then
    systemctl restart tor
fi
exit 0
```

```
mkdir -p /var/lib/networkd-dispatcher/routable.d
chmod 755 /var/lib/networkd-dispatcher/routable.d/00-restart-tor
chown root /var/lib/networkd-dispatcher/routable.d/00-restart-tor
```

When using `systemd-networkd` the script gets exectued everytime the interface `wlan0` has a successfull reconnect.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onny** [comment]
Not sure what the ofborg error means :(

**Comment by SuperSandro2000** [comment]
Looks like the daemon died

@ofborg eval

File: pkgs/tools/networking/networkd-dispatcher/default.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    install -Dm644 networkd-dispatcher.service $out/lib/systemd/system/networkd-dispatcher.service
+    install -Dm644 networkd-dispatcher.conf $out/etc/conf.d/networkd-dispatcher.conf
+    install -D networkd-dispatcher.8 -t $out/share/man/man8/
+    runHook postInstall
+  '';
+
+  doCheck = true;
+
+  checkPhase = ''
+    pytest tests
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/networkd-dispatcher/default.nix:60]: Please use pytestCheckHook

File: nixos/modules/services/networking/networkd-dispatcher.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+with lib;
+
+let
+  cfg = config.services.networkd-dispatcher;
+in {
+  options = {
+    services.networkd-dispatcher = {
+      enable = mkEnableOption (mdDoc ''
+        Networkd-dispatcher service for systemd-networkd connection status
+        change. See [https://gitlab.com/craftyguy/networkd-dispatcher](upstream instructions)
```
Comment (SuperSandro2000) [inline nixos/modules/services/networking/networkd-dispatcher.nix:12]: We should at least add some generic option where to write the settings. Note: nixos has no /usr/, so using your example as is will not work.

**Comment by onny** [comment]
Thank you for the review @SuperSandro2000 :)

**Comment by happysalada** [comment]
Thank you !

**Pull request merged**